program/c/src/oracle/sort/test_sort_stable.c

```diff
@@ -1,73 +0,0 @@
-#include <stdio.h>
-#include "../util/util.h"
-
-#define BEFORE(i,j) (((i)>>16)<((j)>>16))
-
-#define SORT_NAME        sort
-#define SORT_KEY_T       int
-#define SORT_IDX_T       int
-#define SORT_BEFORE(i,j) BEFORE(i,j)
-#include "tmpl/sort_stable.c"
-
-int test_sort_stable() {
-
-# define N 96
-  int x[N];
-  int y[N];
-  int w[N];
-
-  /* Brute force validate small sizes via the 0-1 principle (with
-     additional information in the keys to validate stability as well). */
-
-  for( int n=0; n<=24; n++ ) {
-    for( long b=0L; b<(1L<<n); b++ ) {
-      for( int i=0; i<n; i++ ) x[i] = (((int)((b>>i) & 1L))<<16) | i;
-      for( int i=0; i<n; i++ ) w[i] = x[i];
-
-      int * z = sort_stable( x,n, y );
-
-      /* Make sure that z is a permutation of input data */
-      for( int i=0; i<n; i++ ) {
-        int j = z[i] & (int)0xffff; /* j is the index where z was initially */
-        if( j<0 || j>=n || z[i]!=w[j] ) { printf( "FAIL (corrupt)\n" ); return 1; }
-        w[j] = -1; /* Mark that this entry has already been confirmed */
-      }
-      for( int i=0; i<n; i++ ) if( w[i]!=-1 ) { printf( "FAIL (perm)\n" ); return 1; }
-
-      /* Make sure that z is in order and stable */
-      for( int i=1; i<n; i++ )
-        if( z[i]<=z[i-1] ) { printf( "FAIL (%s, b=%lx)\n", BEFORE( z[i], z[i-1] ) ? "order" : "stable", b ); return 1; }
-    }
-  }
```

Comment:
This test was very nice, can we do something similar.
(Testing the price_model on all arrays with 0 or 1 as values).

Reply:
Added.